src/Bicep.Cli.IntegrationTests/BuildCommandTests.cs

```diff
@@ -182,6 +182,36 @@ public async Task Build_Valid_SingleFile_After_Restore_Should_Succeed(DataSet da
                 actualLocation: compiledFilePath);
         }
 
+        [DataTestMethod]
+        [DynamicData(nameof(GetParamData), DynamicDataSourceType.Method, DynamicDataDisplayNameDeclaringType = typeof(DataSet), DynamicDataDisplayName = nameof(DataSet.GetDisplayName))]
+        public async Task Build_Valid_Params_File_Should_Succeed(DataSet dataSet)
+        {
+        
+            var outputDirectory = dataSet.SaveFilesToTestDirectory(TestContext);
+            var paramsFilePath = Path.Combine(outputDirectory, DataSet.TestFileMainParam);
+
+            var (output, error, result) = await Bicep("build", paramsFilePath);
+
+            using (new AssertionScope())
+            {
+                result.Should().Be(0);
+                output.Should().BeEmpty();
+                AssertNoErrors(error);
+            }
+
+            var compiledFilePath = Path.Combine(outputDirectory, DataSet.TestFileMainCompiled);
+            File.Exists(compiledFilePath).Should().BeTrue();
+
+            string content = File.ReadAllText(compiledFilePath);
+            var actual = JToken.Parse(content);
+
+            actual.Should().EqualWithJsonDiffOutput(
+                TestContext,
+                JToken.Parse(dataSet.Compiled!),
+                expectedLocation: Path.Combine("src", "Bicep.Core.Samples", "Files", dataSet.Name, DataSet.TestFileMainCompiled),
```

Comment:
Does this even compare the paths? because I am emtting to main.bicepparam.json not main.json but the test is still passing

Reply:
Both `compiledFilePath` (L202) and this `expectedLocation` property use `DataSet.TestFileMainCompiled` as the file basename, so if the test is compiling to `main.bicepparam.json`, then the test will be checking at `main.bicepparam.json`, too.